MASFoundation/Classes/_private_/services/model/MASModelService.m

```diff
@@ -1991,6 +1993,30 @@ - (void)loginAsRefreshTokenWithCompletion:(MASCompletionErrorBlock)completion
                                           //
                                           if (error)
                                           {
+                                              //
+                                              // DE509848 - During server maintenance timeframe,
+                                              // when token refresh calls returns 500, MAS library logout the user.
+                                              //
+                                              if ([MAS isDonotLogoutTokenRenewalOnServerErrors] && httpResponse.statusCode == 500)  {
+                                                  
+                                                  NSError *idTokenValidationError = nil;
+                                                  BOOL isIdTokenValid = [MASAccessService validateIdToken:[[MASAccessService sharedService] getAccessValueStringWithStorageKey:MASKeychainStorageKeyIdToken]
+                                                                                            magIdentifier:[[MASAccessService sharedService] getAccessValueStringWithStorageKey:MASKeychainStorageKeyMAGIdentifier]
+                                                                                                    error:&idTokenValidationError];
+                                                  
+                                                  NSString *refreshToken =
+                                                    [MASAccessService sharedService].currentAccessObj.refreshToken;
+                                                  if (refreshToken && isIdTokenValid && !idTokenValidationError)
+                                                  {
+                                                      if (blockCompletion) {
+                                                      
+                                                          blockCompletion(NO, error);
+                                                      }
+                                                  
+                                                      return;
+                                                  }
```

Comment:
[minor] please stick to the one style of the indentation. Remove unnecessary empty lines.

Reply:
Done.